homeassistant/components/filesize/sensor.py

```diff
@@ -79,13 +74,13 @@ class FilesizeEntity(CoordinatorEntity[FileSizeCoordinator], SensorEntity):
     def __init__(
         self,
         description: SensorEntityDescription,
-        path: str,
         entry_id: str,
         coordinator: FileSizeCoordinator,
     ) -> None:
         """Initialize the Filesize sensor."""
         super().__init__(coordinator)
-        base_name = path.split("/")[-1]
+        assert coordinator.path  # Can not be None if coordinator is initialized
```

Comment:
I think there is something wrong. Considering the comment and the place this is added (entity constructor).

The assert is here, because it cannot be `None`, however, the comment says it can be?

Which one is it? If it can be `None`, it should be handled. If it can't be `None`, it should not be able to be `None`.

Reply:
Yeah. Mistake with words.
It can not be `None` as the coordinator **is** initialized and thereby we have a path.
I'll fix 👍

Reply:
Converted the PR back to draft, as the issue is not resolved.

The wording change still highlights the same issue. If it cannot be `None` then the assert makes no sense and should be fixed.

Reply:
Perhaps we read it differently. The assert is there to tell mypy it's not `None` even if it's initialized as such in the coordinator but later updated on first refresh. 
I'll remove the assert and just ignore the mypy error instead. I think that probably makes better sense.

Reply:
> The assert is there to tell mypy it's not None even if it's initialized as such in the coordinator but later updated on first refresh.

I get that... so this assert is here wrongly, it indicates another issue (incorrect typing upstream).

Reply:
Don't agree it's wrong but let's change the logic so the issue goes away.